### PR TITLE
Fix connection of animated nodes and scroll offset with useNativeDriver.

### PR DESCRIPTION
**Pull request by msand**
Add example showing regression before this fix is applied.

## Summary

https://github.com/facebook/react-native/pull/18187 Was found to introduce a regression in some internal facebook code-base end to end test which couldn't be shared. I was able to create a reproducible demo of a regression I found, and made a fix for it. Hopefully this will fix the internal test, such that the pr can stay merged.

## Changelog

[GENERAL] [Fixed] - Fix connection of animated nodes and scroll offset with useNativeDriver.

## Test Plan

The commit includes a new example for RNTester. Unit and E2E integration could be added in a later commit.

ping @cpojer

### CODE REVIEW

**Comment by cpojer** [comment]
Let me get back to this one!

File: Libraries/NativeAnimation/Nodes/RCTInterpolationAnimatedNode.m [hunk 1]
```diff
@@ -61,11 +123,47 @@ - (void)performUpdate
...
+                                                          _extrapolateRight);
+        NSString *str;
+        if (_shouldRound) {
+          // rgba requires that the r,g,b are integers.... so we want to round them, but we *dont* want to
+          // round the opacity (4th column).
+          bool isAlpha = i == 3;
+          CGFloat rounded = isAlpha ? round(val * 1000) / 1000 : round(val);
+          str = isAlpha ? [NSString stringWithFormat:@"%1.3f", rounded] : [NSString stringWithFormat:@"%1.0f", rounded];
+        } else {
+          str = [NSString stringWithFormat:@"%1f", val];
```
Comment (matpaul) [inline Libraries/NativeAnimation/Nodes/RCTInterpolationAnimatedNode.m]: After check example with arc https://github.com/react-native-community/react-native-svg/issues/951
found issue with fixed rounding, 
proposal to fix 
```
NSNumber *numberValue = [NSNumber numberWithDouble:val];
str = [numberValue stringValue];
```

**Comment by matpaul** [comment]
@cpojer Hello! what stop to merge this pr?)
Tested it with patch in production everything work very good (morphing 'svg' with nativeDriver awesome!)

**Comment by cpojer** [comment]
I finally got a strong enough computer to run the tests locally at FB :D I rebased this internally at FB and can verify that as far as I can see the previous issue should be resolved. I'm trying to land it now.

**Comment by msand** [comment]
@cpojer I made a small change to optimize the connection queue handling

**Pull request closed without merging**